pandas/_libs/missing.pyx

```diff
@@ -434,6 +449,34 @@ class NAType(C_NAType):
 
     __rxor__ = __xor__
 
+    # What else to add here? datetime / Timestamp? Period? Interval?
```

Comment:
Timestamp et al don't implement `__array_ufunc__` right? If they would, I would say they could handle NA instead of NA handling all those types.. 

(I also think it is not that important right now that pd.NA can handle those types, since we don't yet use NA for those dtypes)